src/mlpack/bindings/go/mlpack/capi/cli_util.cpp

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file cli_util.cpp
+ * @author Yasmine Dumouchel
+ * @author Yashwant Singh
+ *
+ * Utility function for Go to set and get parameters to and from the CLI.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/bindings/go/mlpack/capi/cli_util.h>
+#include "cli_util.hpp"
+#include <mlpack/core/util/cli.hpp>
+
+namespace mlpack {
+
+extern "C" {
+
+/**
+ * Set the double parameter to the given value.
+ */
+void mlpackSetParamDouble(const char *identifier, double value)
+{
+  util::SetParam(identifier, value);
+}
+
+/**
+ * Set the int parameter to the given value.
+ */
+void mlpackSetParamInt(const char *identifier, int value)
+{
+  util::SetParam(identifier, value);
+}
+
+/**
+ * Set the float parameter to the given value.
+ */
+void mlpackSetParamFloat(const char *identifier, float value)
+{
+  util::SetParam(identifier, value);
+}
+
+/**
+ * Set the bool parameter to the given value.
+ */
+void mlpackSetParamBool(const char *identifier, bool value)
+{
+  util::SetParam(identifier, value);
+}
+
+/**
+ * Set the string parameter to the given value.
+ */
+void mlpackSetParamString(const char *identifier, const char *value)
+{
+  std::string val;
+  val.assign(value);
+  util::SetParam(identifier, val);
```

Comment:
I have tried this but this doesn't work.

Reply:
Ahh, yeah, I think it's because `SetParam()` takes a reference, not a const reference, so we have to create a `std::string`.  I'll go ahead and resolve this; thanks for trying though. :+1: